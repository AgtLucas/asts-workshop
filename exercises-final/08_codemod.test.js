import fs from 'fs'
import * as babel from 'babel-core'
import * as recast from 'recast'
import nodeESModulePlugin from './08_codemod'

const esmFixFixture = require.resolve('./__testfixtures__/esm-fix/main.js')

const esmFixContent = fs.readFileSync(esmFixFixture, 'utf8')

test('codemods imports of CommonJS modules', () => {
  const {code} = babel.transform(esmFixContent, {
    filename: esmFixFixture,
    parserOpts: {parser: recast.parse},
    generatorOpts: {generator: recast.print},
    babelrc: false,
    plugins: [nodeESModulePlugin],
  })
  expect(code).toMatchSnapshot()
})

test('I submitted my elaboration and feedback', () => {
  const submitted = true
  expect(true).toBe(submitted)
})

//////// EXTRA CREDIT ////////

// If you get this far, try adding a few more tests,
// then file a pull request to add them to the extra credit!
// Learn more here: http://kcd.im/asts-workshop-contributing
