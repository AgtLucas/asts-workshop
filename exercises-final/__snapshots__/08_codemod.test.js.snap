// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`codemods imports of CommonJS modules 1`] = `
"import fs from 'fs';
import _path from 'path';
import isAbsolute from './cjs';
import cjs2 from './cjs2';
import myFunction, { someProp, otherFunction } from './both';
import { foo } from './esm';
import * as esm2 from './esm2';
import defaultExport from './default-esm';

function doThings(withThing) {
  if (foo()) {
    const resolve = 'I am not the resolve function!';
    return fs.writeFileSync('./foo.ignored.txt', resolve);
  } else if (defaultExport() && isAbsolute(withThing)) {
    const foo = 'not the foo function';
    return myFunction(_path.resolve(foo, someProp), otherFunction);
  } else if (cjs2.foobar()) {
    return cjs2.baz.buzz;
  } else if (esm2.default) {
    return { foo: esm2.foo };
  }
  return isAbsolute.theAnswer;
}

doThings('/blah/blah');"
`;
